java/clients/src/main/java/sleeper/status/report/ingest/job/IngestQueueMessages.java

```diff
@@ -0,0 +1,161 @@
+import sleeper.configuration.properties.InstanceProperty;
+import sleeper.job.common.QueueMessageCount;
+
+import java.io.PrintStream;
+import java.util.Objects;
+import java.util.Optional;
+
+import static sleeper.configuration.properties.SystemDefinedInstanceProperty.BULK_IMPORT_EKS_JOB_QUEUE_URL;
+import static sleeper.configuration.properties.SystemDefinedInstanceProperty.BULK_IMPORT_EMR_JOB_QUEUE_URL;
+import static sleeper.configuration.properties.SystemDefinedInstanceProperty.BULK_IMPORT_PERSISTENT_EMR_JOB_QUEUE_URL;
+import static sleeper.configuration.properties.SystemDefinedInstanceProperty.INGEST_JOB_QUEUE_URL;
+
+public class IngestQueueMessages {
+    private final Integer ingestMessages;
+    private final Integer emrMessages;
+    private final Integer persistentEmrMessages;
+    private final Integer eksMessages;
+
+    private IngestQueueMessages(Builder builder) {
+        ingestMessages = builder.ingestMessages;
+        emrMessages = builder.emrMessages;
+        persistentEmrMessages = builder.persistentEmrMessages;
+        eksMessages = builder.eksMessages;
+    }
+
+    public static IngestQueueMessages from(InstanceProperties properties, QueueMessageCount.Client client) {
+        return builder()
+                .ingestMessages(getMessages(properties, client, INGEST_JOB_QUEUE_URL))
+                .emrMessages(getMessages(properties, client, BULK_IMPORT_EMR_JOB_QUEUE_URL))
+                .persistentEmrMessages(getMessages(properties, client, BULK_IMPORT_PERSISTENT_EMR_JOB_QUEUE_URL))
+                .eksMessages(getMessages(properties, client, BULK_IMPORT_EKS_JOB_QUEUE_URL))
+                .build();
+    }
+
+    private static Integer getMessages(
+            InstanceProperties properties, QueueMessageCount.Client client, InstanceProperty property) {
+        String queueUrl = properties.get(property);
+        if (queueUrl == null) {
+            return null;
+        } else {
+            return client.getQueueMessageCount(queueUrl).getApproximateNumberOfMessages();
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public void print(PrintStream out) {
+        int queuesEnabled = 0;
+        if (ingestMessages != null) {
+            queuesEnabled++;
+            out.printf("Jobs waiting in ingest queue (excluded from report): %s%n", ingestMessages);
+        }
+        if (emrMessages != null) {
+            queuesEnabled++;
+            out.printf("Jobs waiting in EMR queue (excluded from report): %s%n", emrMessages);
+        }
+        if (persistentEmrMessages != null) {
+            queuesEnabled++;
+            out.printf("Jobs waiting in persistent EMR queue (excluded from report): %s%n", persistentEmrMessages);
+        }
+        if (eksMessages != null) {
+            queuesEnabled++;
+            out.printf("Jobs waiting in EKS queue (excluded from report): %s%n", eksMessages);
+        }
+        if (queuesEnabled > 1) {
```

Comment:
Should this be > 0?

Reply:
Then you'd get something like this:

Jobs waiting in ingest queue (excluded from report): 10
Total jobs waiting across all queues: 10

In that case the second line doesn't seem to add much value.

Reply:
True, but it might be better to be consistent and always log the total jobs waiting?